diff --git a/launchmon/src/linux/lmon_api/lmon_coloc_spawner.cxx b/launchmon/src/linux/lmon_api/lmon_coloc_spawner.cxx
index e43bbd0..79e4bb9 100644
--- a/launchmon/src/linux/lmon_api/lmon_coloc_spawner.cxx
+++ b/launchmon/src/linux/lmon_api/lmon_coloc_spawner.cxx
@@ -122,4 +122,4 @@ spawner_coloc_t::do_frontend()
       memcpy(lmonpl, (*iter).c_str(), (*iter).size() + 1);
       lmonpl += (*iter).size() + 1;
     }
-  lmonpl = '\0'; /* ending null */
+  *lmonpl = '\0'; /* ending null */

   if (write_lmonp_long_msg(m_be_master_sockfd, msg, msgsize) < 0)
     {
