diff --git a/configure.ac b/configure.ac
index 0663fbd..4ab5603 100644
--- a/configure.ac
+++ b/configure.ac
@@ -294,7 +294,7 @@ echo --------
 # check in supplibs directory 
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_HEADER(readline/readline.h,
   [ AC_CHECK_HEADER(readline/history.h,
     [ AC_CHECK_LIB(readline,tgetent,
@@ -310,7 +310,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
       ])
     ])
   ])
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 fi
 # check in system locations
 if test "$use_readline" != "yes" -a "$ga_dyn_supplibs" = "yes" ; then
@@ -334,7 +334,7 @@ echo -------
 # check in supplibs directory 
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir
-  GA_SET_FLAGS([],[],[-ltiff])
+#  GA_SET_FLAGS([],[],[-ltiff])
   AC_CHECK_HEADER(geotiff.h,
   [ AC_CHECK_HEADER(geotiffio.h,
     [ AC_CHECK_HEADER(xtiffio.h, 
@@ -350,7 +350,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
       ])
     ])
   ])
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 fi
 # check in system locations
 if test "$use_geotiff" != "yes" -a "$ga_dyn_supplibs" = "yes" ; then
@@ -381,7 +381,7 @@ echo ----------
 # check in supplibs directory 
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_HEADER([shapefil.h], 
   [ AC_CHECK_LIB([shp], [main], 
     [ have_shp=yes 
@@ -390,7 +390,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
       use_shp=yes 
     ])
   ])    
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 fi
 # check in system locations
 if test "$use_shp" != "yes" -a "$ga_dyn_supplibs" = "yes" ; then
@@ -413,63 +413,63 @@ fi
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir for additional dependent libs
 
-  # png15
+  # png16
   echo 
   echo png
   echo ---
-  GA_SET_FLAGS([])
-  AC_CHECK_LIB([png15], [main], [have_png=yes])
-  GA_UNSET_FLAGS
+#  GA_SET_FLAGS([])
+  AC_CHECK_LIB([png16], [main], [have_png=yes])
+#  GA_UNSET_FLAGS
 
   # zlib
   echo 
   echo zlib
   echo ----
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_HEADER([zlib.h], 
   [ AC_CHECK_LIB([z], [compress], [have_zlib=yes])
   ])
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 
   # jpeg
   echo 
   echo jpeg
   echo ----
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_LIB([jpeg], [main], [have_jpeg=yes])
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 
   # jasper
   echo 
   echo jasper
   echo ------
-  GA_SET_FLAGS([jasper])
+#  GA_SET_FLAGS([jasper])
   AC_CHECK_HEADER([jasper.h],
   [ AC_CHECK_LIB([jasper], [main], [have_jasper=yes])
   ])
-  GA_UNSET_FLAGS
+#  GA_UNSET_FLAGS
 
   # xml2
   echo 
   echo xml2
   echo ----
   if test "$have_zlib" = "yes" ; then
-    GA_SET_FLAGS([libxml2],[],[],[-lz])
+#    GA_SET_FLAGS([libxml2],[],[],[-lz])
     AC_CHECK_LIB([xml2], [main], [have_xml2=yes])
-    GA_UNSET_FLAGS
+#    GA_UNSET_FLAGS
   fi
 
   # curl (needed for netcdf4 and gadap)
   echo 
   echo curl
   echo ----
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_LIB([curl], [main], 
   [ have_curl=yes 
     # this will be used to check if curl is https-enabled
     CURL_CONFIG=${ga_supplib_dir}/bin/curl-config
   ])
-  GA_UNSET_FLAGS	
+#  GA_UNSET_FLAGS
 
 fi
 
@@ -481,7 +481,7 @@ echo --
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir 
   if test "$have_png" = "yes" -a "$have_zlib" = "yes" -a "$have_jpeg" = "yes" ; then
-    GA_SET_FLAGS([])
+#    GA_SET_FLAGS([])
     AC_CHECK_HEADER(gd.h,
     [ AC_CHECK_LIB([gd], [gdImageCreate], 
       [ AC_CHECK_LIB([gd], [gdImageJpeg], 
@@ -491,7 +491,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
         ])
       ])
     ])
-    GA_UNSET_FLAGS
+#    GA_UNSET_FLAGS
   fi 
 fi
 # check in system locations
@@ -521,15 +521,15 @@ echo -----
 if test "Z$ga_supplib_dir" != "Z" ; then 
   if test "$have_zlib" = "yes" -a "$have_png" = "yes" -a "$have_jasper" = "yes" ; then
   echo looking in $ga_supplib_dir 
-    GA_SET_FLAGS([])
+#    GA_SET_FLAGS([])
     AC_CHECK_HEADER(grib2.h,
     [ AC_CHECK_LIB([grib2c], [main], 
       [ use_grib2=yes 
 #        GA_SET_LIB_VAR([grib2_libs], [grib2c jasper png15 z])
-        GA_SET_DYNLIB_VAR([grib2_libs], [grib2c jasper png15 z])
+        GA_SET_DYNLIB_VAR([grib2_libs], [grib2c jasper png16 z])
       ])
     ])
-    GA_UNSET_FLAGS
+#    GA_UNSET_FLAGS
   fi
 fi
 # check in system locations
@@ -557,7 +557,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
   if test "$have_xml2" = "yes" -a  "$have_zlib" = "yes" -a "$have_png" = "yes" ; then
     echo looking in $ga_supplib_dir 
     # check for required header files 
-    GA_SET_FLAGS([cairo freetype2])
+    # GA_SET_FLAGS([cairo freetype2])
     AC_CHECK_HEADER(cairo.h,
     [ AC_CHECK_HEADER(cairo-xlib.h,
     [ AC_CHECK_HEADER(ft2build.h,
@@ -579,7 +579,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
     ])
     ])
     ])
-    GA_UNSET_FLAGS
+    # GA_UNSET_FLAGS
   fi
 fi
 # check in system locations
@@ -608,7 +608,7 @@ echo ----
 if test "Z$ga_supplib_dir" != "Z" ; then 
   if test "$have_zlib" = "yes" ; then
     echo looking in $ga_supplib_dir 
-    GA_SET_FLAGS([])
+#    GA_SET_FLAGS([])
     AC_CHECK_HEADER([hdf5.h],
     [ AC_CHECK_LIB([hdf5], [main], 
       [ AC_CHECK_LIB([hdf5_hl], [main], 
@@ -619,7 +619,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
         ])
       ])
     ])
-    GA_UNSET_FLAGS
+#    GA_UNSET_FLAGS
   fi
 fi
 # check in system locations
@@ -647,13 +647,13 @@ echo udunits
 echo -------
 if test "Z$ga_supplib_dir" != "Z" ; then 
   echo looking in $ga_supplib_dir 
-  GA_SET_FLAGS([])
+#  GA_SET_FLAGS([])
   AC_CHECK_HEADER([udunits.h], 
-  [ AC_CHECK_LIB([udunits], [utInit], 
+  [ AC_CHECK_LIB([udunits2], [utInit], 
     [ have_udunits=yes 
     ])
   ])
-  GA_UNSET_FLAGS	
+#  GA_UNSET_FLAGS
 fi
 # check in system locations
 if test "$have_udunits" = "no" -a "$ga_dyn_supplibs" = "yes" ; then
@@ -673,7 +673,7 @@ echo ----
 if test "Z$ga_supplib_dir" != "Z" ; then 
   if test "$have_zlib" = "yes" -a "$have_jpeg" = "yes" -a "$have_udunits" = "yes" ; then
     echo looking in $ga_supplib_dir 
-    GA_SET_FLAGS([])
+#    GA_SET_FLAGS([])
     AC_CHECK_HEADER([mfhdf.h],
     [ AC_CHECK_LIB([df], [main], 
       [ AC_CHECK_LIB([mfhdf], [main],
@@ -676,7 +676,7 @@
       [ AC_CHECK_LIB([mfhdf], [main],
         [ use_hdf=yes
 #          GA_SET_LIB_VAR([hdf_libs], [mfhdf df jpeg z udunits])
-          GA_SET_DYNLIB_VAR([hdf_libs], [mfhdf df jpeg z udunits])
+          GA_SET_DYNLIB_VAR([hdf_libs], [mfhdf df jpeg z udunits2])
         ])
       ])
     ])
@@ -683,7 +683,7 @@ if test "Z$ga_supplib_dir" != "Z" ; then
         ])
       ])
     ])
-    GA_UNSET_FLAGS
+#    GA_UNSET_FLAGS
   fi
 fi
 # check in system locations
@@ -714,7 +714,7 @@ echo ------
 if test "$have_udunits" = "yes" ; then
   if test "Z$ga_supplib_dir" != "Z" ; then
     echo looking in $ga_supplib_dir 
-    GA_SET_FLAGS([])
+#    GA_SET_FLAGS([])
     AC_CHECK_HEADER([netcdf.h],
     [ AC_CHECK_LIB([netcdf], [main],
       [ use_nc="yes"
@@ -723,7 +723,7 @@ if test "$have_udunits" = "yes" ; then
 	if test ! -f "$NC_CONFIG" || test ! -x "$NC_CONFIG"; then
           use_nc4="no"
 #          GA_SET_LIB_VAR(nc_libs, [udunits netcdf])
-          GA_SET_DYNLIB_VAR(nc_libs, [udunits netcdf])
+          GA_SET_DYNLIB_VAR(nc_libs, [udunits2 netcdf])
         else
           # check if netcdf lib has hdf5 and if dap-enabled
           nchdf5=`$NC_CONFIG --has-hdf5`
@@ -731,7 +731,7 @@ if test "$have_udunits" = "yes" ; then
           if test "$nchdf5" = "yes" -a "$have_curl" = "yes" ; then
             use_nc4="yes"
 #            GA_SET_LIB_VAR(nc_libs, [udunits netcdf hdf5_hl hdf5 curl z])
-            GA_SET_DYNLIB_VAR(nc_libs, [udunits netcdf hdf5_hl hdf5 curl z])
+            GA_SET_DYNLIB_VAR(nc_libs, [udunits2 netcdf hdf5_hl hdf5 curl z])
           fi
         fi
       ])
@@ -787,7 +787,7 @@ if test "x$with_gadap" != xno ; then
   if test "Z$ga_supplib_dir" != "Z" ; then 
     if test "$have_curl" = "yes" -a "$have_xml2" = "yes" -a "$have_zlib" = "yes" ; then
       echo looking in $ga_supplib_dir 
-      GA_SET_FLAGS([],[],[],[-lz])
+#      GA_SET_FLAGS([],[],[],[-lz])
       AC_CHECK_HEADER(gadap.h,
       [ AC_CHECK_LIB(gadap,main,
         [ AC_CHECK_LIB(dap,main,
@@ -799,7 +799,7 @@ if test "x$with_gadap" != xno ; then
           ])
         ])
       ])
-      GA_UNSET_FLAGS
+#      GA_UNSET_FLAGS
     fi
   fi
 # check in system locations
