diff --git a/src/MountPointAttr.h b/src/MountPointAttr.h
index 16e3eb4..c0a6f4f 100644
--- a/src/MountPointAttr.h
+++ b/src/MountPointAttr.h
@@ -237,9 +237,9 @@ namespace FastGlobalFileStatus {
     };
 
 
-    const int BASE_FS_SPEED = 1;
-    const int BASE_FS_SCALABILITY = 1;
-    const int INDIRECTION = -1;
+    const short unsigned int BASE_FS_SPEED = 1;
+    const short unsigned int BASE_FS_SCALABILITY = 1;
+    const short unsigned int INDIRECTION = 0xFFFFu;
 
 
     /**
